pytest/tests/mocknet/helpers/neard_runner.py

```diff
@@ -785,9 +795,10 @@ def check_genesis_state(self):
         if not running:
             logging.error(
                 f'neard exited with code {exit_code} on the first run')
-            # For now just exit, because if this happens, there is something pretty wrong with
+            # For now just set the state to ERROR, because if this happens, there is something pretty wrong with
             # the setup, so a human needs to investigate and fix the bug
-            sys.exit(1)
+            self.set_state(TestState.ERROR)
+            self.save_data()
```

Comment:
Similar question here. Is it ok not to take lock when saving data?

Reply:
same as above^